iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -355,6 +355,16 @@ def sort_nodes(self):
         node_order = toposort_flatten(node_deps)
         self.nodes = [self.nodes[x] for x in node_order]
 
+        #Check inputs all in the beginning
+        root_node_section_passed = False
+        for node in self.nodes:
+            if not root_node_section_passed:
+                if 'input' not in str(node):
+                    root_node_section_passed = True
+            else:
+                if 'input' in str(node):
+                    raise NodeConnectionError("Inputs not sorted in the beginning")
```

Comment:
You should include the node and its position as keyword arguments in this error so its easy to identify which node caused the problem.